python3Packages.python-mapnik: fix build on x86_64

Dettorer opened the pull request:
###### Description of changes

For Python3.10: disable the OGR test, which fails since a gdal update at af10ec1c5b7af7106a5b5ea2b633275d549d7e8c (Upstream tests are still considered unreliable according to https://github.com/mapnik/python-mapnik/issues/255).

For Python3.11: use the 1.82 version of boost instead of the default 1.79. The import checks didn't pass because of a boost binding problem solved by https://github.com/boostorg/python/pull/385.

ZHF: #230712

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - the few non-disabled tests of the upstream testsuite.
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

###### nixpkgs-review output

Ran `nix run '.#nixpkgs-review' -- rev HEAD --print-result`

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.python-mapnik</li>
    <li>python310Packages.python-mapnik.dist</li>
    <li>python311Packages.python-mapnik</li>
    <li>python311Packages.python-mapnik.dist</li>
  </ul>
</details>

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Dettorer committed:
python310Packages.python-mapnik: disable the OGR test

The OGR test fails since a gdal update at
af10ec1c5b7af7106a5b5ea2b633275d549d7e8c. Upstream tests are still
considered unreliable according to
https://github.com/mapnik/python-mapnik/issues/255.

Dettorer committed:
python311Packages.python-mapnik: use boost182 instead of the default

The import checks didn't pass because of a boost binding problem solved
by https://github.com/boostorg/python/pull/385.

Dettorer commented:
@erictapen hi! I see you were the maintainer of this package in the past, would you still have time to review this fix? I also see you removed yourself from the maintainers list at some point so I'd understand if you didn't want to spend time on this. I'll ask someone else in a few days if needed and won't ping you again.

mweinelt commented:
Can you check the ofborg logs of the darwin builds? They have some broken tests, maybe they are salvagable.

Dettorer committed:
python3Packages.python-mapnik: fix tests on darwin

- replace the hardcoded references to /tmp with $TMPDIR
- disable a PDF-related cairo test

Dettorer commented:
Done, the tests were storing temporary files in hardcoded `/tmp` paths, which doesn't seem writable in the darwin nix sandbox, I added a pre-check step that replaces all occurrences of `/tmp` in the test files with `$TMPDIR`. I also disabled another test that was failing exclusively on darwin, I guess the discrepancy is caused by a different cairo version but I didn't really investigate.
